includes/model.php

```diff
@@ -749,6 +752,16 @@ public static function get_image_by_url( string $url = '' ): ?int {
 		$id   = isset( $results[0]->ID ) ? absint( $results[0]->ID ) : 0;
 		$guid = $results[0]->guid ?? null;
 
+		/**
+		 * Filter the image ID found by ISC
+		 *
+		 * Especially useful for compatibility with plugins that manipulate the image URL or meta data so that our checks don’t work
+		 *
+		 * @param int   $id     attachment ID or null
+		 * @param array $params array with additional, optional parameters
+		 */
+		$id = apply_filters( 'isc_filter_final_id_get_image_by_url', $id, [ 'original_url' => $original_url, 'newurl' => $newurl, 'url' => $url ] );
```

Comment:
**Suggestion:** It seems like you're using the variable `$newurl` in the `apply_filters` function call at line 763, but it's not defined anywhere in the provided code. Make sure to define `$newurl` before using it. [bug]
```suggestion
  $newurl = <some_value>; // Define $newurl here
  $id = apply_filters( 'isc_filter_final_id_get_image_by_url', $id, [ 'original_url' => $original_url, 'newurl' => $newurl, 'url' => $url ] );
```